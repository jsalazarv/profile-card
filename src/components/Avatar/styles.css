.avatar {
    @apply flex;
    @apply items-center;
    @apply justify-center;
    @apply overflow-auto;
}
