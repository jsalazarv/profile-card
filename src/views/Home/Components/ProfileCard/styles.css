.profile-card {
    @apply flex;
    @apply justify-center;
    @apply items-center;
    @apply w-96;
    @apply p-4;
    @apply bg-white dark:bg-gray-400;
    @apply border border-blue-100 dark:border-amber-500;
    @apply transition-colors duration-1000;
    @apply shadow-2xl;
}

.profile-card__avatar {
    @apply border-4;
    @apply border-r-black dark:border-r-amber-500;
    @apply border-l-black dark:border-l-amber-500;
    @apply transition-colors duration-1000;
}

.profile-card__content {
    @apply mt-4;
}

.profile-card__content h1 {
    @apply text-lg;
    @apply font-bold;
    @apply text-black dark:text-amber-500;
    @apply transition-colors duration-1000;
}

.profile-card__content h3 {
    @apply text-base;
    @apply text-center;
    @apply text-black dark:text-white;
    @apply transition-colors duration-1000;
}