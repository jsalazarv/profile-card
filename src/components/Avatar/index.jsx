import './styles.css';
import { useMemo } from 'react';

export const Avatar = ({
  className = '',
  children,
  size = 'small',
  type = 'circle',
  color = 'bg-primary',
}) => {
  const sizes = {
    small: 'w-6 h-6',
    medium: 'w-16 h-16',
    large: 'w-32 h-32',
  };

  const types = {
    circle: 'rounded-full',
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };

  const avatarSize = useMemo(() => sizes[size], [size]);
  const avatarType = useMemo(() => types[type], [type]);

  return (
    <div
      className={`overflow-auto ${avatarSize} ${avatarType} ${color} ${className}`}>
      {children}
    </div>
  );
};
