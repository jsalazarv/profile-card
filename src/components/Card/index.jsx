import './styles.css';
import { useMemo } from 'react';

export const Card = ({
  className = '',
  children,
  type = 'tile',
  outlined = '',
}) => {
  const types = {
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };
  const outline = 'border-solid border';
  const cardType = useMemo(() => types[type], [type]);

  return (
    <div className={`card ${cardType} ${outlined && outline} ${className}`}>
      {children}
    </div>
  );
};
