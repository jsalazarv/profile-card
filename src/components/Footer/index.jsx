import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import './styles.css';

export const Footer = ({ phrase = 'Designed with' }) => {
  return (
    <>
      <footer className="footer">
        <p>
          {phrase}
          <span>
            <FontAwesomeIcon icon={faHeart} />
          </span>
        </p>
      </footer>
    </>
  );
};
