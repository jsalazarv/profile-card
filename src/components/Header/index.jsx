import './styles.css';
import { ThemeToggle } from '../ThemeToggle/index.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

export const Header = ({}) => {
  return (
    <header className="header">
      <nav className="header__navbar" aria-label="Top">
        <div className="navbar__content">
          <div className="logo__container">
            <a href="https://jsalazarv.dev/" target="_blank">
              <h2>[Js]</h2>
            </a>
          </div>
          <div className="items__container">
            <a href="https://github.com/jsalazarv/profile-card" target="_blank">
              <FontAwesomeIcon
                className="text-2xl border-black dark:text-amber-500 dark:text-3xl transition-all duration-1000"
                icon={faGithub}
              />
            </a>
            <ThemeToggle />
          </div>
        </div>
      </nav>
    </header>
  );
};
