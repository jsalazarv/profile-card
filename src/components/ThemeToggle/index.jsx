import './styles.css';
import useDarkMode from '../../hooks/useDarkMode.js';

export const ThemeToggle = ({ className = '' }) => {
  const [colorTheme, setTheme] = useDarkMode();

  const toggleTheme = () => {
    setTheme(colorTheme);
  };

  return (
    <input
      className={`toggle ${className}`}
      type="checkbox"
      onClick={toggleTheme}
      aria-label="toggleTheme"
    />
  );
};
