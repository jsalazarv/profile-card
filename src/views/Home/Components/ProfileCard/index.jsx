import './styles.css';
import { Card } from '../../../../components/Card/index.jsx';
import { Avatar } from '../../../../components/Avatar/index.jsx';

export const ProfileCard = () => {
  return (
    <Card className="profile-card" type="rounded">
      <Avatar className="profile-card__avatar" type="circle" size="large">
        <img
          className="grayscale dark:grayscale-0"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQGMsJNRv8en8a8mnoT_W9D7CjqV55SAPTurg&usqp=CAU"
          alt=""
        />
      </Avatar>
      <div className="profile-card__content">
        <h1>The Notorious B.I.G.</h1>
        <h3>United States</h3>
      </div>
      <div className="profile-card__footer"></div>
    </Card>
  );
};
