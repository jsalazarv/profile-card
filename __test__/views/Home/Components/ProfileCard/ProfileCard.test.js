import { render } from '@testing-library/react';
import { ProfileCard } from '../../../../../src/views/Home/Components/ProfileCard';
import { Avatar } from '../../../../../src/components/Avatar';

jest.mock('../../../../../src/components/Avatar');

describe('[ProfileCard Component]', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Should content [profile-card] class', () => {
    const { container } = render(<ProfileCard />);

    expect(container.firstChild).toHaveClass('profile-card');
    expect(container).toMatchSnapshot();
  });

  test('Should content [Avatar] component', () => {
    render(<ProfileCard />);

    expect(Avatar).toHaveBeenCalledOnce();
  });
});
