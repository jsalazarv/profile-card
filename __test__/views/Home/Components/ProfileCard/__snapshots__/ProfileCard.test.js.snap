// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ProfileCard Component] Should content [profile-card] class 1`] = `
<div>
  <div
    class="card rounded-lg  profile-card"
  >
    <div
      class="profile-card__content"
    >
      <h1>
        The Notorious B.I.G.
      </h1>
      <h3>
        United States
      </h3>
    </div>
    <div
      class="profile-card__footer"
    />
  </div>
</div>
`;
